### PR TITLE
TFLu: Pointwise subtraction Int8

### PR DESCRIPTION
**Pull request by giuseros**

### CODE REVIEW

**Comment by gbaned** [comment]
@giuseros Could you please resolve the conflicts? Thanks!

File: tensorflow/lite/experimental/micro/kernels/sub.cc [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/kernels/op_macros.h"
+
+namespace tflite {
+namespace ops {
+namespace micro {
+namespace sub {
+
+namespace {
+
```
Comment (petewarden) [inline tensorflow/lite/experimental/micro/kernels/sub.cc]: Can you put these math functions in a new file in tensorflow/lite/kernels/internal/reference/sub.h, remove the duplicate code from tensorflow/lite/kernels/internal/reference/reference_ops.h, and #include the file at the top of reference_ops.h?

That will help us use the same code from TFL mobile and micro, so we aren't increasing the duplication.

**Review by petewarden — CHANGES_REQUESTED** [review]

File: tensorflow/lite/experimental/micro/kernels/sub.cc (hunk 1 above)
Comment (giuseros) [reply tensorflow/lite/experimental/micro/kernels/sub.cc]: Hi Peter, 
The reason I didn't do this, is because there is a ScopedProfilingLabelWrapper definition in (most of) the reference_ops.h reference kernels which does not compile in micro. Do you have any suggestion about how can I fix it? Should I remove the definition?

Thanks

**Comment by gbaned** [comment]
@petewarden Can you please take a look at this PR? Thanks!

**Comment by gbaned** [comment]
@giuseros Could you please resolve the conflicts? Thanks!

File: tensorflow/lite/experimental/micro/kernels/sub.cc (hunk 1 above)
Comment (giuseros) [reply tensorflow/lite/experimental/micro/kernels/sub.cc]: Hi Pete, 
I was wondering if you had any update about this PR. 

Thank you a lot

File: third_party/mlir/BUILD [hunk 2]
```diff
@@ -2456,12 +2456,13 @@ filegroup(
 
 exports_files(
     [
-        "include/mlir/Dialect/StandardOps/Ops.td",
```
Comment (njeffrie) [inline third_party/mlir/BUILD]: I'm confused why these changes are necessary for this CL?  Did this make it in by accident or did I miss something?

File: tensorflow/lite/experimental/micro/kernels/sub.cc (hunk 1 above)
Comment (njeffrie) [reply tensorflow/lite/experimental/micro/kernels/sub.cc]: We recently added ruy::profiler::ScopeLabel label(); from  tensorflow/lite/experimental/ruy/profiler/instrumentation.h.  This #ifdefs this label to a no-op if RUY_PROFILER is not defined, which makes it work on micro.

Please create tensorflow/lite/kernels/internal/reference/sub.h like Pete mentioned, but use ruy::profiler::ScopeLabel label();

**Comment by giuseros** [comment]
Hi @petewarden @njeffrie @freddan80,
I tried to apply your fixes: could you review this please?

**Comment by giuseros** [comment]
Hi @petewarden @njeffrie @freddan80,
Any update on this?

**Review by njeffrie — DISMISSED** [review]

**Review by njeffrie — DISMISSED** [review]

**Comment by gbaned** [comment]
@giuseros Can you please check build failures? Thanks!

**Comment by gbaned** [comment]
@giuseros Can you please address Ubuntu Sanity errors? Thanks!

**Review by njeffrie — APPROVED** [review]
